web/src/app/workplace/taskdetails/taskdetails.component.ts

```diff
@@ -46,4 +46,12 @@ export class TaskdetailsComponent implements OnInit {
   openTask(taskId: string) {
     this.router.navigate([{outlets: {detail: `task/${taskId}`}}], {relativeTo: this.route.parent});
   }
+
+  workOnTaskDisabled(): boolean {
+    return this.task === null ? false : this.task.state === 'COMPLETED';
```

Comment:
Do not compare with null in typescript or javascript, there is no need for that and some bugs can come from that
return this.task? this.task.state === 'COMPLETED' : false ;